BLUESPAWN-client/src/util/eventlogs/EventLogs.cpp

```diff
@@ -9,291 +9,196 @@ const int ARRAY_SIZE = 10;
-					if (!query.SearchesByValue())
-						params.push_back(query.ToString());
-				if (ERROR_SUCCESS != (status = EventToEventLogItem(hEvents[i], &detect, params)))
-					goto cleanup;
+		DWORD dwReturned{};
+		while(EvtNext(hResults, ARRAY_SIZE, hEvents, INFINITE, 0, &dwReturned)){
+			for(DWORD i = 0; i < dwReturned; i++) {
 
-				results.push_back(detect);
+				auto item = EventToEventLogItem(hEvents[i], params);
+				if(item){
+					results.push_back(*item);
+				}
 
 				EvtClose(hEvents[i]);
 				hEvents[i] = NULL;
-
 			}
 		}
 
-	cleanup:
-
-		for (DWORD i = 0; i < dwReturned; i++)
-			if (NULL != hEvents[i])
+		for(unsigned i = 0; i < ARRAY_SIZE; i++){
+			if(hEvents[i]){
 				EvtClose(hEvents[i]);
+			}
+		}
+
+		if(GetLastError() != ERROR_NO_MORE_ITEMS){
+			LOG_ERROR("EventLogs::ProcessResults: EvtNext failed with " << GetLastError());
+		}
 
 		return results;
 	}
 
-	DWORD EventLogs::EventToEventLogItem(EVT_HANDLE hEvent, EventLogItem* pItem, std::vector<std::wstring>& params) {
+	std::optional<EventLogItem> EventToEventLogItem(const EventWrapper& hEvent, const std::vector<std::wstring>& params){
 		DWORD status = ERROR_SUCCESS;
 
-		std::wstring eventIDStr;
-		std::wstring eventRecordIDStr;
-		std::wstring timeCreated;
-		std::wstring channel;
-		std::wstring rawXML;
-
-		if (ERROR_SUCCESS != (status = GetEventParam(hEvent, &eventIDStr, L"Event/System/EventID")))
-			return status;
-		if (ERROR_SUCCESS != (status = GetEventParam(hEvent, &eventRecordIDStr, L"Event/System/EventRecordID")))
-			return status;
-		if (ERROR_SUCCESS != (status = GetEventParam(hEvent, &timeCreated, L"Event/System/TimeCreated/@SystemTime")))
-			return status;
-		if (ERROR_SUCCESS != (status = GetEventParam(hEvent, &channel, L"Event/System/Channel")))
-			return status;
-		if (ERROR_SUCCESS != (status = GetEventXML(hEvent, &rawXML)))
-			return status;
+		std::optional<std::wstring> eventIDStr, eventRecordIDStr, timeCreated, channel, rawXML;
+
+		if (std::nullopt == (eventIDStr = GetEventParam(hEvent, L"Event/System/EventID")))
+			return std::nullopt;
+		if (std::nullopt == (eventRecordIDStr = GetEventParam(hEvent, L"Event/System/EventRecordID")))
+			return std::nullopt;
+		if (std::nullopt == (timeCreated = GetEventParam(hEvent, L"Event/System/TimeCreated/@SystemTime")))
+			return std::nullopt;
+		if (std::nullopt == (channel = GetEventParam(hEvent, L"Event/System/Channel")))
+			return std::nullopt;
+		if (std::nullopt == (rawXML = GetEventXML(hEvent)))
+			return std::nullopt;
+
+		EventLogItem pItem{};
 
 		// Provide values for filtered parameters
 		for (std::wstring key : params) {
-			std::wstring val;
-			if (ERROR_SUCCESS != (status = GetEventParam(hEvent, &val, key))) {
-				LOG_ERROR(L"EventLogs::EventToDetection: Failed query parameter " + key + L" with code " + std::to_wstring(status));
-				return status;
+			std::optional<std::wstring> val = GetEventParam(hEvent, key);
+			if (!val) {
+				return std::nullopt;
 			}
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'status' is assigned a value that is never used.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=5011005)